### PR TITLE
refactor: 💡 application serializeAttribute

### PR DESCRIPTION
**Pull request by DhariniJeeva**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-10195)

Refactor application attribute serializer method to include a check for `for` option and delete attributes not matched to that specific record type.

### CODE REVIEW

File: addons/api/addon/serializers/application.js [hunk 1]
```diff
@@ -72,6 +73,16 @@ export default class ApplicationSerializer extends RESTSerializer {
       }
       delete json[key];
     }
+    // If an attribute has a `for` option, it must match the
+    // record's `type`, else the attribute is excluded
+    // from serialization.
+    if (options?.for && !options.for.includes(recordType)) {
```
Comment (ZedLi) [inline addons/api/addon/serializers/application.js:79]: Can we actually be more specific than using `includes` for strings and do a real equality check? This works fine if `for` is an array but for strings I'm concerned that in the future we can have bugs where we have a type that is a substring of another type.

**Review by ZedLi — COMMENTED** [review]
Can we also add some tests for this functionality?

File: addons/api/addon/serializers/application.js (hunk 1 above)
Comment (DhariniJeeva) [reply addons/api/addon/serializers/application.js]: Good point, Done!

**Review by lisbet-alvarez — APPROVED** [review]

**Review by dvonanderson — APPROVED** [review]

File: addons/api/addon/serializers/application.js [hunk 2]
```diff
@@ -72,6 +73,20 @@ export default class ApplicationSerializer extends RESTSerializer {
       }
       delete json[key];
     }
+    // If an attribute has a `for` option, it must match the
+    // record's `type`, else the attribute is excluded
+    // from serialization.
```
Comment (ZedLi) [inline addons/api/addon/serializers/application.js:78]: We should also probably be able to handle secrets but I don't think we need to worry about it yet, maybe just make a note in the comment that we don't handle it yet and can add support if/when we need it?

**Review by ZedLi — APPROVED** [review]
Looks good!

**Review by cameronperera — APPROVED** [review]

**Pull request merged**